src/presign/participant.rs

```diff
@@ -182,7 +238,7 @@ impl PresignParticipant {
         &mut self,
         rng: &mut R,
         message: &Message,
-        input: &Storage,
+        input: &Input,
```

Comment:
thought: with the current architecture, it doesn't seem necessary to validate the input (because it gets constructed correctly in `construct_presign_input`), but I think we'll want to do so after #205 is complete. I'll make a separate issue to add validation checks to externally-provided inputs so that we spot issues with it as early as possible (e.g. instead of running half the protocol before realizing you don't have enough `KeySharePublic`s).